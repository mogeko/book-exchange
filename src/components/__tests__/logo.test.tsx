import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Logo from "@/components/logo";
import logoImage from "@/public/logo.svg";

describe("Logo", () => {
  it("Create a text logo", () => {
    const { container } = render(<Logo>Test Logo</Logo>);
    expect(container).toBeInTheDocument();
    expect(container.querySelector("h1")?.textContent).toEqual("Test Logo");
    expect(container.querySelector("a")?.href).toMatch(/([&/]$)/);
    expect(container.querySelector("img")).toBeNull();
  });

  it("snapshot a text logo", () => {
    const { container } = render(<Logo>Test Logo</Logo>);
    expect(container).toMatchSnapshot();
  });

  it("Create an image logo", () => {
    const { container } = render(<Logo src={logoImage} />);
    expect(container).toBeInTheDocument();
  });
});
