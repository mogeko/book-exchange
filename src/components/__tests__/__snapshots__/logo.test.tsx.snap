// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshot a text logo 1`] = `
<div>
  <div
    class="flex justify-start"
  >
    <a
      class="btn btn-ghost normal-case"
      href="/"
    >
      <span
        class="sr-only"
      >
        Test Logo
      </span>
      <h1
        class="text-xl font-bold text-base-content"
      >
        Test Logo
      </h1>
    </a>
  </div>
</div>
`;
